source/github/github.go

```diff
@@ -43,27 +44,30 @@ type Config struct {
 }
 
 func (g *Github) Open(url string) (source.Driver, error) {
+	var client *http.Client
 	u, err := nurl.Parse(url)
 	if err != nil {
 		return nil, err
 	}
 
 	if u.User == nil {
-		return nil, ErrNoUserInfo
-	}
-
-	password, ok := u.User.Password()
-	if !ok {
-		return nil, ErrNoUserInfo
-	}
+		// use http.DefaultClient
+		client = nil
```

Comment:
Doesn't `client` default to `nil`? Invert the condition and don't bother with an `else` branch.

Reply:
Done.